zerver/tests/test_auth_backends.py

```diff
@@ -747,3 +748,162 @@ def test_fetch_auth_backend(self):
                 'dev': True,
                 'result': 'success',
             })
+
+class TestDevAuthBackend(ZulipTestCase):
+    def test_login_success(self):
+        # type: () -> None
+        email = 'hamlet@zulip.com'
+        user_profile = get_user_profile_by_email(email)
+        data = {'direct_email': email}
+        result = self.client_post('/accounts/login/local/', data)
+        self.assertEqual(result.status_code, 302)
+        self.assertEqual(get_session_dict_user(self.client.session), user_profile.id)
+
+    def test_login_failure(self):
+        # type: () -> None
+        email = 'hamlet@zulip.com'
+        data = {'direct_email': email}
+        with self.settings(AUTHENTICATION_BACKENDS=('zproject.backends.EmailAuthBackend',)):
+            with self.assertRaisesRegexp(Exception, 'Direct login not supported.'):
+                self.client_post('/accounts/login/local/', data)
```

Comment:
We should also add a test like this:
```
+    def test_login_nonexistant(self):
+        # type: () -> None
+        email = 'hamlet-does-not-exist@zulip.com'
+        data = {'direct_email': email}
+        with self.assertRaisesRegexp(Exception, 'User cannot login'):
+            self.client_post('/accounts/login/local/', data)
```

For some reason, we seem to be getting failures / extra traceback output with `assertRaisesRegexp` on Python 3.  Any idea what the issue is there?